net_io.c

```diff
@@ -776,13 +782,270 @@ static void send_sbs_heartbeat(struct net_service *service)
     completeWrite(service->writer, data + len);
 }
 
+//
+//=========================================================================
+//
+// Write Stratux output to TCP clients
+//
+static void modesSendStratuxOutput(struct modesMessage *mm, struct aircraft *a) {
+    char *p;
+    struct timespec now;
+    struct tm    stTime_receive, stTime_now;
+    int          msgType;
+
+    // We require a tracked aircraft for SBS output
+    if (!a)
+        return;
+
+    // Don't ever forward 2-bit-corrected messages via SBS output.
+    if (mm->correctedbits >= 2)
+        return;
+
+    // Don't ever forward mlat messages via SBS output.
+    if (mm->source == SOURCE_MLAT)
+        return;
+
+    // Don't ever send unreliable messages via SBS output
+    if (!mm->reliable && !a->reliable)
+        return;
+
+    //// For now, suppress non-ICAO addresses
+    //if (mm->addr & MODES_NON_ICAO_ADDRESS)
+    //    return;
+
+    p = prepareWrite(&Modes.stratux_out, 1000); // larger buffer size needed vs SBS
+    if (!p)
+        return;
+
+    // NOTE: not sure about message types, so keep the original code from
+    // stratux/dump1090 as-is.
+    //
+    // Decide on the basic SBS Message Type
+    if        ((mm->msgtype ==  4) || (mm->msgtype == 20)) {
+        msgType = 5;
+    } else if ((mm->msgtype ==  5) || (mm->msgtype == 21)) {
+        msgType = 6;
+    } else if ((mm->msgtype ==  0) || (mm->msgtype == 16)) {
+        msgType = 7;
+    } else if  (mm->msgtype == 11) {
+        msgType = 8;
+    } else if ((mm->msgtype != 17) && (mm->msgtype != 18)) {
+        return;
+    } else if ((mm->metype >= 1) && (mm->metype <=  4)) {
+        msgType = 1;
+    } else if ((mm->metype >= 5) && (mm->metype <=  8)) {
+        if (mm->cpr_decoded)
+            {msgType = 2;}
+        else
+            {msgType = 7;}
+    } else if ((mm->metype >= 9) && (mm->metype <= 18)) {
+        if (mm->cpr_decoded)
+            {msgType = 3;}
+        else
+            {msgType = 7;}
+    } else if ((mm->metype == 29) || (mm->metype == 31)) {
+        msgType = 9; // new message type for target state and operational status messages
+    } else if (mm->metype !=  19) {
+        return;
+    } else if ((mm->mesub == 1) || (mm->mesub == 2)) {
+        msgType = 4;
+    } else {
+        return;
+    }
```

Comment:
It would be good if this code can go away. As far as I can tell, the SBS message type is not actually used in stratux (the only reference to it I can find is in a CSV export, nothing in the main code?); so either dropping the field entirely or just putting a dummy value in there would be preferable to copy&pasting the SBS output code.